administrator/components/com_media/resources/scripts/components/browser/items/item.js

```diff
@@ -37,14 +37,63 @@ export default {
             return store.state.selectedItems.some(selected => selected.path === item.path);
         }
 
+	    /**
+	     * Create and dispatch onMediaFileSelected Event
+	     *
+	     * @param {object}  data  The data for the detail
+	     *
+	     * @returns {void}
+	     */
+        function sendEvent(data) {
+	        const ev = new CustomEvent('onMediaFileSelected', {"bubbles":true, "cancelable":false, "detail": data});
+	        window.parent.document.dispatchEvent(ev);
+        }
         /**
          * Handle the click event
          * @param event
          */
         function handleClick(event) {
-            var e = createEvent('onMediaFileSelected');
-            e.item = item;
-            window.parent.document.dispatchEvent(e);
+	        let path = false;
+	        const data = {
+		        path: path,
+		        thumb: false,
+		        fileType: item.mime_type ? item.mime_type : false,
+		        extension: item.extension ? item.extension : false,
+	        };
+
+            if (item.type === 'file') {
+	            const csrf = Joomla.getOptions('com_media').csrfToken;
+	            const apiBaseUrl = Joomla.getOptions('com_media').apiBaseUrl;
+	            Joomla.request({
+		            url: `${apiBaseUrl}&task=api.files&url=true&path=${item.path}&${csrf}=1`,
+		            method: 'GET',
+		            perform: true,
+		            headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
+		            onSuccess: (response) => {
+			            const resp = JSON.parse(response);
+			            if (resp.success === true) {
+				            if (resp.data[0].url) {
+					            if (/local-/.test(item.path)) {
```

Comment:
we can have infinite root folders for image storages.Also default **images** folder can be removed by a user from settings

Reply:
I am not checking against `local-0` so the code will happily execute also for `local-999999999999`

Reply:
@kasvith is it posible to make `thumb_path ` camel case, e.g. `thumbPath` ?

Reply:
Sure can do

Reply:
done in #292

Reply:
Closed #292